src/network/listener.rs

```diff
@@ -40,7 +40,7 @@ pub trait Listener {
         // Open socket
         let mut ret_sock = Socket::new(self.network());
         if let Err(e) = ret_sock.connect(self.peer(), self.port()) {
-            return Err(util::Error::Detail("listener".to_owned(), Box::new(e)));
+            return Err(util::Error::Network(network::Error::Detail("listener".to_owned(), Box::new(e))));
         }
```

Comment:
This whole `if` should be replaced by `ret_sock.connect(self.peer(), self.port())?;` because `ret_sock.connect` already returns a `network::Error`.

Then you can drop the new `self` import at the top of this file.